File: mycollab-mobile/src/main/java/com/esofthead/mycollab/mobile/module/project/view/parameters/MilestoneScreenData.java
package com.esofthead.mycollab.mobile.module.project.view.parameters;

import com.esofthead.mycollab.module.project.domain.Milestone;
import com.esofthead.mycollab.module.project.domain.criteria.MilestoneSearchCriteria;
import com.esofthead.mycollab.vaadin.mvp.ScreenData;

/**
 * 
 * @author MyCollab Ltd.
 * @since 4.5.2
 *
 */
public class MilestoneScreenData {
	public static class Read extends ScreenData<Integer> {

		public Read(Integer params) {
			super(params);
		}
	}

	public static class Edit extends ScreenData<Milestone> {

		public Edit(Milestone params) {
			super(params);
		}
	}

	public static class Add extends ScreenData<Milestone> {

		public Add(Milestone params) {
			super(params);
		}
	}

	public static class Search extends ScreenData<MilestoneSearchCriteria> {

		public Search(MilestoneSearchCriteria params) {
			super(params);
		}
	}
}

File: mycollab-mobile/src/main/scala/com/esofthead/mycollab/mobile/module/project/view/parameters/MilestoneScreenData.scala
package com.esofthead.mycollab.mobile.module.project.view.parameters

import com.esofthead.mycollab.module.project.domain.SimpleMilestone
import com.esofthead.mycollab.module.project.domain.criteria.MilestoneSearchCriteria
import com.esofthead.mycollab.vaadin.mvp.ScreenData

/**
  * @author MyCollab Ltd
  * @since 5.2.5
  */
object MilestoneScreenData {
    class Search(param: MilestoneSearchCriteria) extends ScreenData(param) {}

    class Read(param: Integer) extends ScreenData(param) {}

    class Edit(param: SimpleMilestone) extends ScreenData(param) {}

    class Add(param: SimpleMilestone) extends ScreenData(param) {}
}
